src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const copy = [...this];
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (copy[copy.length - 1] === undefined) {
+      copy[copy.length - 1] = '';
+    }
```

Comment:
it's not clear why did you set this check
for what reason did you check the last element of the array?